src/lib/wallet/WalletConnectClient.js

```diff
@@ -0,0 +1,566 @@
+//@flow
+import { useCallback, useEffect, useState } from 'react'
+import WalletConnect from '@walletconnect/client'
+import web3Utils from 'web3-utils'
+import abiDecoder from 'abi-decoder'
+import Web3 from 'web3'
+import { first, sortBy } from 'lodash'
+import AsyncStorage from '../utils/asyncStorage'
+import { delay } from '../utils/async'
+import api from '../../lib/API/api'
+import logger from '../logger/js-logger'
+import { useSessionApproveModal } from '../../components/walletconnect/WalletConnectModals'
+import { useWallet } from './GoodWalletProvider'
+const log = logger.child({ from: 'WalletConnectClient' })
+
+//TODO:
+//7. cancel tx
+//8. edit gas
+//9. advanced edit tx values/contract call values
+//10. events
+//11. show warning if unable to decode contract call
+/**
+ * Parses the read WalletConnet URI from QR Code.
+ * If not valid, returns null.
+ * If valid, returns the WalletConnet URI.
+ * @param {string} link - receive WalletConnect URI
+ * @returns {string|null} - {link|null}
+ */
+export const readWalletConnectUri = link => {
+  // checks that the link has the expected strings in it
+  const eip1328UriFormat = /wc:[\w\d-]+@\d+\?bridge=.*&key=[a-z0-9]+/
+  const validUri = link.match(eip1328UriFormat)[0]
+
+  if (!validUri) {
+    return null
+  }
+
+  return link
+}
+
+export const getWalletConnectTopic = link => {
+  const eip1328UriFormat = /wc:([\w\d-]+)@\d+\?bridge=.*&key=[a-z0-9]+/
+  const topic = link.match(eip1328UriFormat)[1]
+  return topic
+}
+
+let chainsCache = []
+export const useChainsList = () => {
+  const [chains, setChains] = useState(chainsCache)
+  chainsCache = chains
+  useEffect(() => {
+    if (chainsCache.length) {
+      return
+    }
+    api.getChains().then(data => {
+      const fuse = data.find(_ => _.chainId === 122)
+      fuse.explorers = [
```

Comment:
regarding explorer see my suggesrtons and example here https://github.com/GoodDollar/GoodDAPP/pull/3811#discussion_r923695309
https://github.com/GoodDollar/GoodDAPP/pull/3811#discussion_r923697624

Reply:
done